version.txt

```diff
@@ -5,4 +5,3 @@ VERSION_GIT_DATE        20210107
 
```

Comment:
Is Meson parsing this to obtain the version, like what we had with CMake? Or do we have the version in two different places? If the latter, Meson needs to parse it in this single canonical location.

Reply:
Yes, Meson is [parsing that file](https://github.com/Tachi107/pistache/blob/meson/meson.build#L78) except in when defining the [project version](https://github.com/Tachi107/pistache/blob/meson/meson.build#L4).
I think I have a solution for the project version; Meson 0.57 supports reading a file that contains version information, but it must contain only one line, so it is impossible to use the current `version.txt`.
A possible solution would be to split `version.txt` in two separate files, `version.txt` and `version_soname.txt`; this would make the parsing waay easier (string split based on `.` instead of iterating over a "big" file and searching for substrings like "`VERSION_MAJOR`".
Another approach could be removing the soname version entirely and deal only with a single version number that would express both soname and feature version, but it would have its own drawbacks.